public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -103,7 +103,7 @@ export class ElasticDatasource
     this.withCredentials = instanceSettings.withCredentials;
     this.url = instanceSettings.url!;
     this.name = instanceSettings.name;
-    this.index = instanceSettings.database ?? '';
+    this.index = (instanceSettings.jsonData.index || instanceSettings.database) ?? '';
```

Comment:
nitpick: i recommend using `??` instead of `||` here.
```suggestion
    this.index = (instanceSettings.jsonData.index ?? instanceSettings.database) ?? '';
```

in general, it's nearly always better to use `X ?? Y` instead of `X || Y` to describe the "if X is missing use Y instead".

(these two behave quite similar, but they have different behavior when `X` is something javascript considers `falsy`, but not `null` or `undefined`. so for example `0 || 42` will be `42`, but `0 ?? 42` will be `0`)